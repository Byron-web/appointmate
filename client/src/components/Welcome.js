import React from "react";
import { Button, Container } from "react-bootstrap";
import "./styles.css";
import FeatureCard from "./FeatureCard";

//Welcome view that shows in the beginner of the app when a user hasnt logged in, or if a user logs out.
function Welcome() {
  return (
    <div className="welcome--wrapper">
      <div className="welcome--header">
        <h1>
          Welcome to AppointMate: Your Personal Appointment Management App
        </h1>
        <p>
          AppointMate is a user-friendly web application designed to simplify
          and streamline your appointment management process. Whether you're a
          healthcare professional, a personal service provider, or a small
          business owner, AppointMate is here to help you efficiently manage
          your appointments and provide an exceptional experience to your
          clients.
        </p>
        <div className="welcome-features">
          <h2>Key Features</h2>
          <div className="d-flex justify-content-around">
            <FeatureCard
              title="Effortlessly schedule and manage appointments"
              description="With AppointMate, you can easily schedule and manage your appointments in a streamlined and efficient way."
            />
            <FeatureCard
              title="Reduce administrative tasks and paperwork"
              description="Say goodbye to manual paperwork and administrative tasks. AppointMate automates various processes to save you time and effort."
            />
            <FeatureCard
              title="Keep track of important appointment details"
              description="AppointMate allows you to store and access essential appointment information, such as patient names, reason for appointments, and appointment dates, all in one place."
            />
          </div>
        </div>
        <p>
          With AppointMate, you can focus on delivering the best care and
          services to your patients or clients without the hassle of traditional
          appointment management methods. Get started today and experience the
          convenience and efficiency of AppointMate.
        </p>
      </div>
      <div className="banner-info--wrapper">
        <Container className="banner-info--container">
          <h4>
            Login by clicking the login button on the top left of this page
          </h4>
          <div className="banner-info--divider"></div>
          <div className="banner-info-footer">
            <span className="mx-3">Or</span>
            <Button variant="primary">
              <a
                href="https://github.com/Byron-web/appointmate"
                target="_blank"
              >
                View Code
              </a>
            </Button>
          </div>
        </Container>
      </div>
    </div>
  );
}

export default Welcome;
