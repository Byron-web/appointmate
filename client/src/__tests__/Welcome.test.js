import React from "react";
import renderer from "react-test-renderer";
import Welcome from "../components/Welcome";

test("Welcome component renders correctly", () => {
  const component = renderer.create(<Welcome />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
