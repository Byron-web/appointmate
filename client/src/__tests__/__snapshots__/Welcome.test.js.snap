// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component renders correctly 1`] = `
<div
  className="welcome--wrapper"
>
  <div
    className="welcome--header"
  >
    <h1>
      Welcome
    </h1>
  </div>
  <div
    className="banner-info--wrapper"
  >
    <div
      className="banner-info--container container"
    >
      <h4>
        Login by clicking the login button on the top left of this page
      </h4>
      <div
        className="banner-info--divider"
      />
      <div
        className="banner-info-footer"
      >
        <span
          className="mx-3"
        >
          Or
        </span>
        <button
          className="btn btn-primary"
          disabled={false}
          type="button"
        >
          <a
            href="https://github.com/Byron-web/appointmate"
            target="_blank"
          >
            View Code
          </a>
        </button>
      </div>
    </div>
  </div>
</div>
`;
